pkg/gadgets/internal/histogram/histogram.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package histogram provides a Histogram struct that represents a histogram of
+// the number of events that occurred in each interval. The intervals are powers
+// of two, so the slots array is indexed by the power of two minus one. For
+// example, the first slot is the number of events that occurred in the interval
+// [1, 1], the second slot is [2, 3], the third slot is [4, 7], and so on.
+
+package histogram
+
+import (
+	"fmt"
+	"strings"
+)
+
+type Slot struct {
+	Count         uint64 `json:"count"`
+	IntervalStart uint64 `json:"intervalStart"`
+	IntervalEnd   uint64 `json:"intervalEnd"`
+}
+
+// Histogram represents a histogram of the number of events that occurred in
+// each interval.
+type Histogram struct {
+	Unit  string `json:"unit,omitempty"`
+	Slots []Slot `json:"slots,omitempty"`
+}
+
+// New creates a new Histogram.
+func New(unit string, slots []Slot) *Histogram {
+	return &Histogram{
+		Unit:  unit,
+		Slots: slots,
+	}
+}
```

Comment:
Do we really need this function if Histogram is an exported type?

Reply:
Not really, but it looked weird to me to have `NewFromCSlots` and not `New` 😅. I'm ok removing it.

Reply:
A possibility is to make `parseCSlots` -> `NewSlots()` and remove `New` and `NewFromCSlots`. Then the user could create a histogram as:

```
myHist := &Histogram{
		Unit:  unit,
		Slots: NewSlots(cSlots),
	}
```